app/third_party_search.py

```diff
@@ -29,7 +29,7 @@
 
 class Handler(utils.BaseHandler):
     """Proxy to perform search with third-party search engine."""
-    def get(self):
+    def post(self):
```

Comment:
Since you changed the JS code to send the parameters as JSON in POST body instead, I guess you also need to change parameter extraction logic here e.g., self.params.query?